File: package/umt_python/src/ip/ip_to_binary_string.py
from .ip_to_long import ip_to_long


def ip_to_binary_string(ip: str) -> str:
    """
    Converts an IPv4 address to its binary string representation.

    Args:
        ip: IPv4 address (e.g., "192.168.1.1")

    Returns:
        Binary string representation (32 bits)

    Raises:
        ValueError: If IP address is invalid

    Example:
        >>> ip_to_binary_string("192.168.1.1")
        '11000000101010000000000100000001'
        >>> ip_to_binary_string("0.0.0.0")
        '00000000000000000000000000000000'
    """
    return f"{ip_to_long(ip):032b}"

File: package/umt_rust/src/ip/ip_to_binary_string.rs
use crate::ip::ip_to_long::ip_to_long;

/// Converts an IPv4 address to its binary string representation
///
/// # Arguments
/// * `ip` - IPv4 address (e.g., "192.168.1.1")
///
/// # Returns
/// * `Ok(String)` - Binary string representation (32 bits)
/// * `Err` - If IP address is invalid
///
/// # Examples
/// ```
/// use umt_rust::ip::ip_to_binary_string;
/// assert_eq!(ip_to_binary_string("192.168.1.1").unwrap(), "11000000101010000000000100000001");
/// ```
pub fn ip_to_binary_string(ip: &str) -> Result<String, String> {
    Ok(format!("{:032b}", ip_to_long(ip)?))
}
